verilog/analysis/verilog_project.h

```diff
@@ -165,6 +165,30 @@ class InMemoryVerilogSourceFile : public VerilogSourceFile {
   const absl::string_view contents_for_open_;
 };
 
+// Source file that was already parsed and got its own TextStructure.
+// Doesn't require file-system access, nor create temporary files.
+class ParsedVerilogSourceFile : public VerilogSourceFile {
+ public:
+  // filename can be fake, it is not used to open any file.
+  ParsedVerilogSourceFile(absl::string_view filename,
+                          const verible::TextStructureView& text_structure,
```

Comment:
the ownership of this probably needs to be described in the constructor description, as this expects the text_structure be  available for the lifetime of this object.

I'd suggest to make this a `const verible::TextStructuredView*`, also text_structure_. Passing a pointer makes it more obvious that the object is keeping a reference to it.

(usually, when we pass const references in constructors, a copy is made, so this is what the programmer expects. If we keep a reference to the thing we can make it more visible by passing a pointer)

Reply:
Done, done & done